frontend/src/scenes/events/eventsTableLogic.js

```diff
@@ -48,7 +48,12 @@ export const eventsTableLogic = kea({
         props.key,
 
     actions: () => ({
-        setProperties: (properties) => ({ properties }),
+        setProperties: (properties) => {
+            if (Array.isArray(properties) && properties.length === 0) {
+                return { properties: {} }
+            }
+            return { properties }
+        },
```

Comment:
I'll approve it, but feedback for the next PR: we usually try to keep actions as pure oneliners to make it easy to scan/find/read them. They should pretty much just serialize parameters. More complicated logic and exception handling usually goes inside a listener, but this part is fine here. Just I'd move the `if` into an inline property-level if.